examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2097,645 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
 
 	atomic.StoreInt32(&stop, 1)
 }
+
+// AggregationExamples contains examples of aggregation operations.
+func AggregationExamples(t *testing.T, db *mongo.Database) {
+	ctx := context.Background()
+
+	salesColl := db.Collection("sales")
+	airlinesColl := db.Collection("airlines")
+	airAlliancesColl := db.Collection("air_alliances")
+
+	{
+		// Start Aggregation Example 1
+		docs := []interface{}{
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-02-08T09:00:00.000Z"},
```

Comment:
I'm getting an error in the `AggregationExamples` 3 and 4 when the `group` pipeline stage tries to access the `date` field defined here. It is `can't convert from BSON type object to Date`.

I think this might be the wrong way of defining a date in a `bson.D`. Let me know 👍

Reply:
`$date` is actually extended JSON signaling that this field should be a date. It seems like the test data doesn't any extjson-style keys besides `$date`, so the easiest way to do this would be to make a helper function to convert these strings into a `time.Time` instance:

```
rfc3339Milli := "2006-01-02T15:04:05.999Z07:00"
parsedTime, err := time.Parse(rfc3339Milli, timeString) // timeString is "2018-02-08T09:00:00.000Z" in this case
```

Also, the final format of the document should be:
```
bson.D{
    {"date", parsedTime},t
}
```
Note that the format above doesn't have a `$date` sub-document. That sub-document only exists in the extended JSON version to indicate to drivers that this is a special value that needs customized parsing. There's more information about extended JSON formats at https://github.com/mongodb/specifications/blob/master/source/extended-json.rst.

Reply:
Thank you! Discussed offline: I tried the above method of parsing date strings (with a new helper `parseDate`) into `time.Time` objects but I still get the same error: `can't convert from BSON type object to Date`...

Reply:
Discussed offline: everything is fine, but the collections under test aren't being dropped, so the pipelines were being executed against a collection that had data in the incorrect format. Dropping each collection once when we declare the collection handles should fix it.